File: LinearDSDemo.py
from collections import deque

# Array demo
def array_demo():
    array = [1, 2, 3, 4, 5]
    print("Array:")
    print(" ".join(map(str, array)))
    print()

# Linked List demo
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def linked_list_demo():
    head = Node(1)
    second = Node(2)
    third = Node(3)
    
    head.next = second
    second.next = third
    
    print("Linked List:")
    current = head
    while current:
        print(current.data, end=" ")
        current = current.next
    print("\n")

# Stack demo
def stack_demo():
    stack = []
    stack.append(1)
    stack.append(2)
    stack.append(3)
    
    print("Stack:")
    while stack:
        print(stack.pop(), end=" ")
    print("\n")

# Queue demo
def queue_demo():
    queue = deque()
    queue.append(1)
    queue.append(2)
    queue.append(3)
    
    print("Queue:")
    while queue:
        print(queue.popleft(), end=" ")
    print("\n")

# Main function to run all demos
if __name__ == "__main__":
    array_demo()
    linked_list_demo()
    stack_demo()
    queue_demo()

File: LinearDSDemo.cpp
#include <iostream>
#include <stack>
#include <queue>
using namespace std;

// Array demo
void arrayDemo() {
    int array[] = {1, 2, 3, 4, 5};
    cout << "Array:" << endl;
    for (int i = 0; i < 5; i++) {
        cout << array[i] << " ";
    }
    cout << endl << endl;
}

// Linked List demo
struct Node {
    int data;
    Node* next;
};

void linkedListDemo() {
    Node* head = new Node();
    Node* second = new Node();
    Node* third = new Node();
    
    head->data = 1;
    head->next = second;
    
    second->data = 2;
    second->next = third;
    
    third->data = 3;
    third->next = nullptr;
    
    cout << "Linked List:" << endl;
    Node* current = head;
    while (current != nullptr) {
        cout << current->data << " ";
        current = current->next;
    }
    cout << endl << endl;
}

// Stack demo
void stackDemo() {
    stack<int> stack;
    stack.push(1);
    stack.push(2);
    stack.push(3);
    
    cout << "Stack:" << endl;
    while (!stack.empty()) {
        cout << stack.top() << " ";
        stack.pop();
    }
    cout << endl << endl;
}

// Queue demo
void queueDemo() {
    queue<int> queue;
    queue.push(1);
    queue.push(2);
    queue.push(3);
    
    cout << "Queue:" << endl;
    while (!queue.empty()) {
        cout << queue.front() << " ";
        queue.pop();
    }
    cout << endl << endl;
}

int main() {
    arrayDemo();
    linkedListDemo();
    stackDemo();
    queueDemo();
    
    return 0;
}
